src/game/client/draw/w3dmodeldraw.cpp

```diff
@@ -1154,3 +1179,1914 @@ void W3DModelDrawModuleData::Parse_Condition_State(INI *ini, void *instance, voi
+        }
+    }
+}
+
+void W3DModelDraw::CRC_Snapshot(Xfer *xfer)
+{
+    DrawModule::CRC_Snapshot(xfer);
+}
+
+void W3DModelDraw::Xfer_Snapshot(Xfer *xfer)
+{
+    uint8_t ver = 2;
+    xfer->xferVersion(&ver, 2);
+    DrawModule::Xfer_Snapshot(xfer);
+    RecoilInfo recoilinfo;
+    for (int i = 0; i < WEAPONSLOT_COUNT; i++) {
+        unsigned char size = (unsigned char)m_weaponRecoilInfoVec[i].size();
+        xfer->xferUnsignedByte(&size);
+
+        if (xfer->Get_Mode() == XFER_SAVE) {
+            for (auto &info : m_weaponRecoilInfoVec[i]) {
+                recoilinfo.m_state = info.m_state;
+                xfer->xferUser(&recoilinfo.m_state, sizeof(recoilinfo.m_state));
+                recoilinfo.m_shift = info.m_shift;
+                xfer->xferReal(&recoilinfo.m_shift);
+                recoilinfo.m_recoilRate = info.m_recoilRate;
+                xfer->xferReal(&recoilinfo.m_recoilRate);
+            }
+        } else {
+            m_weaponRecoilInfoVec[i].clear();
+            for (int j = 0; j < size; j++) {
+                xfer->xferUser(&recoilinfo.m_state, sizeof(recoilinfo.m_state));
+                xfer->xferReal(&recoilinfo.m_shift);
+                xfer->xferReal(&recoilinfo.m_recoilRate);
+                m_weaponRecoilInfoVec[i].push_back(recoilinfo);
+            }
+        }
+    }
+
+    unsigned char size = (unsigned char)m_subObjects.size();
+    xfer->xferUnsignedByte(&size);
+    ModelConditionInfo::HideShowSubObjInfo subobj;
+
+    if (xfer->Get_Mode() == XFER_SAVE) {
+        for (auto &info : m_subObjects) {
+            subobj.name = info.name;
+            xfer->xferAsciiString(&subobj.name);
+            subobj.visible = info.visible;
+            xfer->xferBool(&subobj.visible);
+        }
+    } else {
+        m_subObjects.clear();
+        for (int j = 0; j < size; j++) {
+            xfer->xferAsciiString(&subobj.name);
+            xfer->xferBool(&subobj.visible);
+            m_subObjects.push_back(subobj);
+        }
+    }
+
+    if (ver >= 2) {
+        if (xfer->Get_Mode() == XFER_SAVE) {
+            if (m_renderObject != nullptr && m_renderObject->Class_ID() == RenderObjClass::CLASSID_HLOD
+                && m_curState != nullptr && m_curState->m_transition) {
+                HLodClass *hlod = static_cast<HLodClass *>(m_renderObject);
+                float frame;
+                int frames;
+                int mode;
+                float multiplier;
+                HAnimClass *anim = hlod->Peek_Animation_And_Info(frame, frames, mode, multiplier);
+                bool b = anim != nullptr;
+                xfer->xferBool(&b);
+
+                if (anim) {
+                    xfer->xferInt(&mode);
+                    float f = frame / (anim->Get_Num_Frames() - 1);
+                    xfer->xferReal(&f);
+                }
+            } else {
+                bool b = false;
+                xfer->xferBool(&b);
+            }
+        } else {
+            bool b;

```

Comment:
Shouldn't `b` have a default `false` here in case serialize read fails to read value?